hypothesis-python/tests/conjecture/common.py

```diff
@@ -276,13 +325,15 @@ def draw_float_kwargs(
 
 
 @st.composite
-def draw_boolean_kwargs(draw, *, use_forced=False):
+def boolean_kwargs(draw, *, use_forced=False):
     forced = draw(st.booleans()) if use_forced else None
     p = draw(st.floats(0, 1, allow_nan=False, allow_infinity=False))
 
     # avoid invalid forced combinations
-    assume(p > 0 or forced is False)
-    assume(p < 1 or forced is True)
+    if forced is True:
+        assume(p > 0)
+    if forced is False:
+        assume(p < 1)
```

Comment:
```python
p = draw(st.floats(0, 1, exclude_min=forced is True, exclude_max=forced is False))
```
The `allow_*` arguments are redundant with two finite endpoints, and this neatly avoids rejection sampling against the minimal and second-simplest values in the space 🙂

Reply:
nice ty!